### ISSUE TITLE
[Feature] Trace viewer: allow fetching traces from origins that require credentials

### ISSUE THREAD

**Issue by pmattj** [microsoft/playwright#28501]
My team is working to allow our developers to load traces directly in the online [trace viewer](https://trace.playwright.dev/). Our ideal scenario is that a developer could open a test result document and jump directly to the trace in the viewer with a simple click of a link.

Our CI infrastructure uploads traces to a location protected by credentials — to fetch a trace, the request needs to contain valid authentication cookies. These cookies are sent if we open the trace URL in the browser directly, but we've found that the online trace viewer does the trace fetch _without_ cookies, even though the origin sends `Access-Control-Allow-Credentials: true`.[1]

We suspected this could be due to the trace viewer not applying the `credentials` option on the fetch request. We tested this theory by deploying the trace viewer locally and adding a `credentials: 'include'` option on [this line](https://github.com/microsoft/playwright/blob/8f056fbbcebe4dd1f1dcedebb43870b81cc475d1/packages/trace-viewer/src/traceModelBackends.ts#L35), and this allowed our local trace viewer to load the trace.

Our request is for this option to be added upstream. This would allow us to use the Playwright-hosted trace viewer rather than hosting it ourselves, and I suspect it would unblock other customers whose traces are credential-protected (assuming their origins also send the proper `Access-Control-Allow-Origin` and `Access-Control-Allow-Credentials` headers).

[1] https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Credentials

**Comment by dgozman** [comment]
Reopening since the fix was reverted in #29024.

**Comment by mxschmitt** [comment]
Note: For the next time we implement it, we should add a URL parameter like `credentials=include` which only then adds this specific property to Fetch. By default it should not do that.

**Comment by pavelfeldman** [comment]
#### Why was this issue closed?

Thank you for your involvement. This issue was closed due to limited engagement (upvotes/activity), lack of recent activity, and insufficient actionability. To maintain a manageable database, we prioritize issues based on these factors.

If you disagree with this closure, please open a new issue and reference this one. More support or clarity on its necessity may prompt a review. Your understanding and cooperation are appreciated.

**Comment by muhamedkarajic** [comment]
I would also be interested in this, please reopen? This is a fundamenal things missing. It would be great if it would somehow allow credentials to a FTP so I avoid having to host my own midddleware, aldoe even with middleware its stil insecure.
